cpp/src/io/orc/reader_impl.cu

```diff
@@ -633,7 +633,16 @@ table_with_metadata reader::impl::read(int skip_rows, int num_rows, int stripe,
     orc_col_map[col] = column_types.size() - 1;
   }
 
-  if (num_rows > 0 && selected_stripes.size() != 0) {
+  auto is_empty_table = not (num_rows > 0 && selected_stripes.size() != 0);
```

Comment:
Since `is_empty_table` is only used in the `if()`, I think the original `if()` logic was cleaner.

Reply:
I believe the concern was confusion about inverting the condition, and whether readers would understand what it meant. `is_empty_table` informs the reader.

Reply:
I had suggested moving the `else` branch to the top since it was tiny and easy to miss after the long `if` branch